### PR TITLE
M32 fix

### LINKED ISSUE #619: Wrong progress reports
The number of bytes in total is wrong. Maybe some fs problem.

~~~
0 % complete, elapsed time: 1687 s, est time: 1245307818 s
SD printing byte 5652/3735928559
~~~

Also this elapsed time report is wrong. I just let it run for a few seconds.
This reports somehow the full 4GB?

Someone else seeing this?

### PR DESCRIPTION
**Pull request by minad**
Fixes #619

### CODE REVIEW

**Comment by wolfmanjm** [comment]
Did you consider that using seek to get file size is no longer required? how about changing it to get the file size from the opendir instead?

**Comment by minad** [comment]
No, I just wanted this to be fixed. I think this class should be cleaned up so or so. But I am not doing that right now.

**Comment by minad** [comment]
I also don't think that it would make it better here. The issue here is just code repetition.

**Comment by minad** [comment]
@wolfmanjm Ping. What about this issue? Since you touched over 82% of the lines of the file Player.cpp, I think you are the right contact person in this case :)

File: src/modules/utils/player/Player.cpp [hunk 1]
```diff
@@ -135,7 +135,7 @@ void Player::on_gcode_received(void *argument)
                 // get size of file
                 int result = fseek(this->current_file_handler, 0, SEEK_END);
                 if (0 != result) {
-                    this->file_size = 0;
+                    this->file_size = -1;
```
Comment (wolfmanjm) [inline src/modules/utils/player/Player.cpp]: reasons for changing this to -1?

**Comment by wolfmanjm** [comment]
this should be a simple fix, not sure why you also changed the initial file size to -1 it was totally unnecessary IMO.

**Comment by minad** [comment]
Yes there is a reason, just read the commit messages. I want to avoid divisons by zero.

**Comment by wolfmanjm** [comment]
as far as I can see divisions by zero cannot occur as they are tested for, so i still so no reason to set it to -1

**Comment by minad** [comment]
You are right. My commit messages somehow states implicitly that I did not check everywhere ;) However ftell also returns negative values in case of an error. So it is only consistent to set it to -1 in case of an fseek error. This was the reason why I changed it.

If you don't like this particular commit, then cherry-pick please.

**Comment by minad** [comment]
@wolfmanjm I removed the -1 commit for you

**Comment by minad** [comment]
but maybe we should still change to long instead of unsigned long to check for ftell errors?

File: src/modules/utils/player/Player.h [hunk 2]
```diff
@@ -50,8 +50,8 @@ class Player : public Module {
         StreamOutput* suspend_stream;
 
         FILE* current_file_handler;
-        unsigned long file_size, played_cnt;
-        unsigned long elapsed_secs;
+        long file_size, played_cnt;
+        long elapsed_secs;
```
Comment (wolfmanjm) [inline src/modules/utils/player/Player.h]: why is elapsed seconds long now? i mace them unsigned for a reason. i understand why file size is long bit i disagree with that too as the error should be detected at source.

**Comment by minad** [comment]
now?

**Pull request merged**